[UI Tests] Moved the doc for UI tests to the UI Tests folder.

pachlava opened the pull request:
### Description
The documentation for WCiOS UI Tests was located together with the other docs for the repo. I think it makes sense to have the documentation in the place for which this documentation is written (it's done so for [WPiOS](https://github.com/wordpress-mobile/WordPress-iOS/tree/trunk/WordPress/WordPressUITests)) - since the person is more likely to find it there.

However, I do realize that there might be a reason to have the file in the original place too (cc @thehenrybyrd) - so I'd like to hear you thoughts too.

### Testing instructions
- Open [WooCommerceUITests](https://github.com/woocommerce/woocommerce-ios/tree/uitii-move-docs-to-dedicated-folder/WooCommerce/WooCommerceUITests) folder while on this branch, and make sure you see the `README.md` rendered automatically (UPD: and all links are functional).
- The PR does not change any app or tests logic, it just moves/renames a doc file to a different location. Having Buildkite 🟢  is good enough.

pachlava committed:
Moved the doc for UI tests to the UI Tests folder.

mokagio reviewed (commented):
> Open [WooCommerceUITests](https://github.com/woocommerce/woocommerce-ios/tree/uitii-move-docs-to-dedicated-folder/WooCommerce/WooCommerceUITests?rgh-link-date=2022-11-29T09%3A55%3A46Z) folder while on this branch, and make sure you see the README.md rendered automatically.

![image](https://user-images.githubusercontent.com/1218433/204540340-ea833fd6-c25b-49c4-a00a-37a717d467ec.png)

👍 

Also +1 on the rationale. I don't know if there's a prior reason for having the docs all in the same place but, if we were starting from scratch, I too think keeping the readme as close as possible to the code it refers to will help discoverability.

---

A minor note—I tried to follow the links in the documented, and got a 404. Example:

![image](https://user-images.githubusercontent.com/1218433/204540413-ddcf3cfb-24ce-4347-90ae-2a980777621a.png)

I suppose that's due to the relative paths being set for the original location and no longer being valid here.

I wonder whether it might be best to remove the links, or to move to absolute links (e.g. https://github.com/woocommerce/woocommerce-ios/blob/f60ff83b155b3f563ff47fb73aa867d1562990ce/WooCommerce/WooCommerceUITests/Tests/LoginTests.swift or to be always up to date https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/WooCommerceUITests/Tests/LoginTests.swift). From one point of view, it's handy to be able to see the up-to-date source from GitHub, from another, there's always the possibility the links will become out-of-date again, if a file is renamed or removed. Maybe it'd be more robust to explain how to find the screens/flows we refer to and trust the reader to be able to find them themselves. 🤔

pachlava committed:
Updated the (now) broken links in the doc.

pachlava commented:
> A minor note—I tried to follow the links in the documented, and got a 404. Example:

Thank you so much for spotting this, @mokagio, I've totally missed this 🤦. Besides broken links, the doc still contained mentions of CircleCI and an outdated guide for running the tests in Xcode. I've addressed those with 7f11ff0de5cc996f37b2af402cdb5708ce7b4839.

- For the links to UI tests files, I removed them. Since the doc now resides in the UI tests folder, I think it's pretty self-explanatory that tests can be found in the `Tests` folder, and the files themselves have descriptive names.
- For links to app build instructions, Buildkite config, I still kept the link relative by adding one more `../`

Thank you!

thehenrybyrd commented:
In general I'd say it makes sense to have this as a README in the UI Tests folder, as you've done. That does make it easy to find for someone working on UI tests.

I don't know the history behind this, but I do see we have a lot of different documents in the [documentation folder](https://github.com/woocommerce/woocommerce-ios/tree/trunk/docs), so I imagine we put the UI Tests docs there because it was the obvious place to put documentation. I don't see an obvious reason to keep it there, as most of the rest of the documentation is development-related.

One thing I would suggest is updating the [WooCommerce README](https://github.com/woocommerce/woocommerce-ios/blob/trunk/README.md), as it includes a link to UI Tests docs which should be updated.

thehenrybyrd commented:
I was able to open [WooCommerceUITests](https://github.com/woocommerce/woocommerce-ios/tree/uitii-move-docs-to-dedicated-folder/WooCommerce/WooCommerceUITests?rgh-link-date=2022-11-29T09%3A55%3A46Z) folder while on this branch, and the links now load as expected.

thehenrybyrd commented on WooCommerce/WooCommerceUITests/README.md:
```diff
@@ -72,25 +72,25 @@ Note that due to the mock server setup, tests cannot be run on physical devices
+1. Follow the [build instructions](../../README.md#build-instructions) to clone the project, install the dependencies, and open the project in Xcode.
 2. Run `rake mocks` to start a local mock server.
-3. With the `WooCommerce` scheme selected in Xcode, navigate to Product > Test Plan and select `UITests`, or open the Test Navigator and select the `UITests` test plan.
+3. Select the `WooCommerceUITests` scheme in Xcode.
```
I'd suggest leaving most of the original wording. It is clearer about how to select the correct scheme.

pachlava replied:
The reason why I went this way, is because `WooCommerce` scheme has `Unit Tests` test plan selected as a default test plan, so to run the `UI Tests` test plan while having `WooCommerce` scheme selected would require editing the scheme every time.

`WooCommerceUITests` scheme is a dedicated scheme which by default has `UI Tests` test plan activated - so it's a matter of one action VS three actions.

thehenrybyrd reviewed (approved):
Other than updating the UI Tests doc link in the [WooCommerce README](https://github.com/woocommerce/woocommerce-ios/blob/trunk/README.md), and the small nitpick below, I'd say this is ready to go!

pachlava committed:
Fixed a link to UI tests doc in the main README.md

pachlava commented:
Thanks for the review @thehenrybyrd !

> One thing I would suggest is updating the [WooCommerce README](https://github.com/woocommerce/woocommerce-ios/blob/trunk/README.md), as it includes a link to UI Tests docs which should be updated.

Good catch! I've fixed the link in the main [README.md](https://github.com/woocommerce/woocommerce-ios/blob/uitii-move-docs-to-dedicated-folder/README.md) (f913d4fe0f87dde5d4a1060bef497b3ab23b0bc0).
